book/src/the-database/moo-value-types.md

```diff
@@ -90,42 +90,33 @@ Object numbers are always integers.
 
-## Anonymous Object Type
-
-Anonymous Objects are references and do not have an object number. They are created by passing the optional third argument to `create()`. Anonymous objects are automatically garbage collected when there is no longer any references to them (in your code or in properties).
+## Flyweights - lightweight objects
 
-We will go into more detail on Anonymous Objects in the [Working with Anonymous Objects](#working-with-anonymous-objects) section.
+mooR adds a new type called flyweights which are lightweight, garbage collected bundles of a delegate object, attributes ("slots") and arbitrary contents, and so combine features of objects, lists, and maps. Verb calls against flyweights dispatch against the delegate, and property lookups work gainst both the delegate and the flyweight's own set of slots.
 
-## Bool Type
-
-_bools_ are either true or false. Eg: `my_bool = true; my_second_bool = false;`. In MOO `true` evaluates to `1` and `false` evaluates to `0`. For example:
+Their syntax looks like:
 
 ```
-false == 0 evaluates to true
-true  == 1 evaluates to true
-false == 1 evaluates to false
-true  == 0 evaluates to false
-true  == 5 evaluates to false
-false == -43 evaluates to false
+< <delegate>, [ slot -> value, ... ], { contents, ... } >
 ```
 
-## WAIF Type
-
-_WAIFs_ are lightweight objects. A WAIF is a value which you can store in a property or a variable or inside a LIST or another WAIF. A WAIF is smaller in size (measured in bytes) than a regular object, and it is faster to create and destroy. It is also reference counted, which means it is destroyed automatically when it is no longer in use. An empty WAIF is 72 bytes, empty list is 64 bytes. A WAIF will always be 8 bytes larger than a LIST (on 64bit, 4 bytes on 32bit) with the same values in it.
-
-> Note: WAIFs are not truly objects and don't really function like one. You can't manipulate a WAIF without basically recreating a normal object (and then what's the point?). It may be better to think of a WAIF as another data type. It's closer to being a list than it is to being an object. But that's semantics, really.
+But the only mandatory element is the delegate.
 
-WAIFs are smaller than typical objects, and faster to create. A WAIF has two builtin OBJ properties, .class and .owner. A WAIF is only ever going to be 4 bytes larger than a LIST with the same values.
+Examples:
 
-OBJs grow by value_bytes(value) - value_bytes(0) for every property you set (that is, every property which becomes non-clear and takes on its own value distinct from the parent). LISTs and WAIFs both grow by value_bytes(value) for each new list element (in a LIST) or each property you set (in a WAIF). So a WAIF is never more than 4 bytes larger than a LIST which holds the same values, except WAIFs give each value a name (property name) but LISTs only give them numbers.
+```
+< $div_tag, [ class -> "background_div" ], { child_node_a, child_node_b } >
```

Comment:
I guess we'll need to extend the `highlight.js` syntax highlighter for this block to look good :D